library/utility/src/uri/ut_uri.e

```diff
@@ -143,7 +143,9 @@ feature {NONE} -- Implementation
 						if has_query then
 							-- Keep query.
 						else
-							query_item := a_base.query_item
+							if has_query then
+								query_item := a_base.query_item
+							end
```

Comment:
I'm not sure I understand this modification. Why not remove the outer if statement since the inner if-statement will never succeed.